### ISSUE TITLE
Sweeping Strikes only triggers from "Yellow" Attacks, meaning abilities, not autoattacks.

### ISSUE THREAD

**Issue by Hanzemon** [chromiecraft/chromiecraft#3444]
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [ ] Horde

### Content Phase:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [ ] 60

### Current Behaviour

The ability Sweeping Strikes will only trigger it's damage when you land an ability, it will never trigger from regular "white" attakcs.

### Expected Blizzlike Behaviour

Sweeping Strikes will proc from both "white" and "yellow" attacks. Meaning it will proc from both autoattack and abilities such as Whirlwind, Cleave, Mortal Strike, Slam, Revenge, etc.

### Source

![sweeping strikes](https://user-images.githubusercontent.com/32701428/166128301-56f21a77-dc7c-4810-aca0-744d63a7d23d.jpg)

![sweep](https://user-images.githubusercontent.com/32701428/166128315-92ad9d14-e783-4796-b8c7-8395e8b1868c.PNG)

### Steps to reproduce the problem

Activate Sweeping Strikes when you have two targets nearby.
Use Autoattack only
Notice how it doesn't proc
Now use abilities and notice how it does proc.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/59e45c251eb53abb0ed4b37a57254cbdf415e69a

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

**Comment by Cooties-cc** [comment]
Perhaps some more proof on the issue. 
[Here is a PvP video from 8/1/2010](url) which should be during the 3.3.5 patch (june 22, 2010), but before the cata prepatch (oct 12, 2010). 
At the 1:19 mark, he intercepts a warlock with a pet out. He then casts sweeping strikes, and follows with a heroic strike that procs sword specialization (extra normal melee swing). The game registers 2 uses of sweeping strikes, bringing the buff down to 3 charges. The MSBT addon also shows 2 procs of sweeping strike damage (1002, 1154) on his pet. Here is an image from the video to get a clearer view. 
<a href="https://imgur.com/UbwL6ec"><img src="https://i.imgur.com/UbwL6ec.png" title="source: imgur.com" /></a>
This also in part shows that SS can be consumed from extra swings. It also shows that it's a normal extra attack against a 2nd nearby mob, and not based on damage done from main attack (as I saw another closed ticket about SS damage). 

|
|
|


Also not to open another ticket about it, unless it's needed, but it also shows that AoE attacks are only consuming 1 charge of sweeping strikes. At the 1:24 mark, he casts whirlwind, which strikes 2 targets (the warlock, and his pet). However it shows only 1 charge of sweeping was consumed. Image for reference. 
<a href="https://imgur.com/VRuY4iV"><img src="https://i.imgur.com/VRuY4iV.png" title="source: imgur.com" /></a>

Currently on the server, whirlwind damage is consuming sweeping strikes as much targets as it hits. 
<a href="https://imgur.com/wNCpnEe"><img src="https://i.imgur.com/wNCpnEe.png" title="source: imgur.com" /></a>
Whirlwind hits 4, 4 SS charges get consumed. 

This also happens on cleave. 
<a href="https://imgur.com/9uu4UgE"><img src="https://i.imgur.com/9uu4UgE.png" title="source: imgur.com" /></a>
2 cleave strikes, 2 SS consumed. 

Bladestorm seems to be working fine, as each time it hits, only 1 charge is consumed (even if you're hitting max 4 targets). However the other ones need to be fixed, as it's not blizzlike.

**Comment by Gultask** [comment]
Thanks Hanzemon for the report! And thanks Cooties-cc for the additional sources too.

I can confirm and I can also confirm that it is working incorrectly for Cleave as well. Cleave should consume 2 charges of Sweeping Strikes, not just one.

**Comment by Hanzemon** [comment]
> Perhaps some more proof on the issue. [Here is a PvP video from 8/1/2010](https://www.warcraftmovies.com/movieview.php?id=159107) which should be during the 3.3.5 patch (june 22, 2010), but before the cata prepatch (oct 12, 2010). At the 1:19 mark, he intercepts a warlock with a pet out. He then casts sweeping strikes, and follows with a heroic strike that procs sword specialization (extra normal melee swing). The game registers 2 uses of sweeping strikes, bringing the buff down to 3 charges. The MSBT addon also shows 2 procs of sweeping strike damage (1002, 1154) on his pet. Here is an image from the video to get a clearer view. [![](https://camo.githubusercontent.com/87c234d8eccf892f34a31e2a3af0491b2b0b08de3f35a952e2411dc2d6e87c2b/68747470733a2f2f692e696d6775722e636f6d2f5562774c3665632e706e67)](https://imgur.com/UbwL6ec) This also in part shows that SS can be consumed from extra swings. It also shows that it's a normal extra attack against a 2nd nearby mob, and not based on damage done from main attack (as I saw another closed ticket about SS damage).
> 
> Also at 1:27 mark, he does an auto+MS, which registers 2 sweeping charges used. So normal whites (assuming you may think sword specialization was considered a special attack) should be procing SS. [![](https://camo.githubusercontent.com/bb184de005de2efa567958e2ecbe26c4d8b908a07fe3d25ba38e87aa521537b4/68747470733a2f2f692e696d6775722e636f6d2f6278743142316c2e706e67)](https://imgur.com/bxt1B1l)
> 
> | | |
> 
> Also not to open another ticket about it, unless it's needed, but it also shows that whirlwind is only consuming 1 charge of sweeping strikes. At the 1:24 mark, he casts whirlwind, which strikes 2 targets (the warlock, and his pet). However it shows only 1 charge of sweeping was consumed. Image for reference. [![](https://camo.githubusercontent.com/ddd94e6823c826e1d856bb7671cb68d66bc4302ea61589688baa89f6f96118bd/68747470733a2f2f692e696d6775722e636f6d2f565275593469562e706e67)](https://imgur.com/VRuY4iV)
> 
> Currently on the server, whirlwind damage is consuming sweeping strikes as much targets as it hits. [![](https://camo.githubusercontent.com/a4a25471172024f1e77b42aeee1523468aa8eb64fb95072bfdb19442bbf1fa31/68747470733a2f2f692e696d6775722e636f6d2f774e43706e45652e706e67)](https://imgur.com/wNCpnEe) Whirlwind hits 4, 4 SS charges get consumed.
> 
> Bladestorm seems to be working fine, as each time it hits, only 1 charge is consumed (even if you're hitting max 4 targets).

Thank you for complementing my answer, you are correct on everything. 

Something that should not be overlooked is that the Sweeping Strike damage should not be calculated from how much damage the ability or attack that procced did. But it should be it's own source calculated from your weapon damage and stats/bonuses.

**Comment by Gultask** [comment]
>Something that should not be overlooked is that the Sweeping Strike damage should not be calculated from how much damage the ability or attack that procced did. But it should be it's own source calculated from your weapon damage and stats/bonuses.

There's an issue going around (closed for now) about SS damage here: https://github.com/azerothcore/azerothcore-wotlk/issues/7636 but I wouldn't know where to start. I barely got Keira3 working just yesterday, I'm no expert 😅

**Comment by bonfax4** [comment]
https://web.archive.org/web/20110612030818/http://www.wowhead.com:80/spell=12328


> By [ZarDoom](https://web.archive.org/web/20110612030818/http://www.wowhead.com/user=ZarDoom) (2·20) [on 2010/09/07](https://web.archive.org/web/20110612030818/http://www.wowhead.com/spell=12328) (Patch 3.3.5)
> Using cleave while under the effect of Sweeping Strikes will consume two charges, and make two additional attacks. So if you're fighting two ennemies, Sweeping Strikes + Cleave = 2 hits on each.
> 
> Profit.
> 
> That being said, Whirlwind (or one second of bladestorming) will only consume ONE charge per spin, not 4 (maximum number of targets of whilrwind = 4), hitting an additional target per whrilwind, not per ennemy hit by the whirlwind.
> 
> (Sorry for my bad english, I'm really doing my best.)
> Last edited by [ZarDoom](https://web.archive.org/web/20110612030818/http://www.wowhead.com/user=ZarDoom) on 2010/09/11 (Patch 3.3.5)
> 

Now the interaction with execute not exactly sure, [wowwiki ](https://wowwiki-archive.fandom.com/wiki/Sweeping_Strikes?direction=prev&oldid=2378895)says execute dmg is not transfered if the target isn´t < 20% health , but the wowhead i linked above does say execute dmg is cleaved but it cleaves the damage as if it was a normal atack, wich seems to be how sweeping seemed to work since vanilla ( [Patch 1.1](https://wowwiki-archive.fandom.com/wiki/Patch_1.10.0#Warriors) - "Execute - This ability will now work with Sweeping Strikes again. If the second victim is below 20% health, they will be hit with the full Execute amount. If the second victim is not below 20% health, they will be hit with normal melee swing damage." ).

**Comment by bonfax4** [comment]
Also going through that linked video it did seem to show only specials were being cleaved. there were a few auto atacks on that warlock that were not. It's also possible being in a ramp made the positional requirements wonky, i've seen blade furry work strangely as well.

**Comment by Cooties-cc** [comment]
The notes seem to be misleading or outdated on the wiki page about the spell, or at least the wording is a tad odd. Like it wasn't getting updated as other information was. If you notice on the patch notes for the spell, patch 2.3 increased the number of extra strikes to 10 from 5, but never documented that it was reverted to 5, down from 10. I assume it was during the 3.0.2 patch when deathwish and sweeping swapped places again, that the 10 charges was changed too. We can clearly see in the pvp video, it is 5 charges, not 10. 
We also see the wordage under MS and Exe to be misleading too. For Exe, it says "not cause [Execute] to strike a nearby opponent". This doesn't mean that the Exe spell is being cast twice, but rather the extra normal hit from Exe while SS is up will not be used if the 2nd target is not below 20% too. Basically not a Exe+Exe, but Exe+normal swing. MS is the same way too, saying " In the case of Mortal Strike, though, only the damage is dealt; the second mob will not get the debuff from the attack.". This is being said in a way that assumes MS is being cast twice, which it is not. We can see from the PvP video that when MS is cast, the 2nd mob struck with the SS charge, is taking a normal swing in dmg. Not a 2nd MS cast. 
Essentially the wiki seems confusing in their wording, making it seem like SS is copying the spell that consumed the charge. When really, all charges consumed for SS are always normal melee swings. 

@bonfax4 It was 100% due to positioning of the players. I posted a screenshot clearly showing that MS + auto was consuming 2 charges of sweeping strikes. It's not possible that SS "sometimes" consumes a charge with a normal melee swing. It will always consume a charge, while the requirements are met (2nd player nearby being the main issue).

**Comment by Cooties-cc** [comment]
I checked the video you linked at 3:46, and I don't think that was a sword spec proc, but rather the last tick of bladestorm. 
I also noticed during the same video that around 3:27 a charge is consumed, as the 2nd hit strikes the paladin running by for 1380 dmg (it was proced from a 3147 dmg mortal strike). You may also notice the paladin was behind him at the time the damage was registered, which could mean that the range of the sweeping strike is 360 degrees around the caster, but only extends outwards of their max melee range. 
<a href="https://imgur.com/f1n0FmL"><img src="https://i.imgur.com/f1n0FmL.png" title="source: imgur.com" /></a>
Given the video recording, it could be safe to assume that SS is omnidirectional, with the only requirement being a 2nd body within range of melee striking distance. Not a 180 frontal cone.

**Comment by bonfax4** [comment]
The white dmg you see during a bladestorm can only be a sword spec proc, and the charge got consumed right when it happened. YEAH latency could easily have a factor into those positional issues. The guy keeps sweeping strikes up for most of the video and you don't see that happen at all for the most part unless someone is the "frontal cone".

**Comment by bonfax4** [comment]
https://www.mmo-champion.com/threads/2022048-Sweeping-strikes-and-ww 
Interesting read in here aswell because this same sweeping strikes id (26654) they claim that works specifically for bladestorm and whirlwind exists in the [Wotlk db](https://wotlkdb.com/?search=Sweeping+strikes#abilities) along with another one.

**Comment by Cooties-cc** [comment]
The mmo-champ forum posts are from 2016 (2016-07-05). So it can't be taken as any evidence for mechanics. Tracking back the same spell ID leads it to be an NPC ability in [classic] vanilla, used by 4 NPCs here : https://classic.wowhead.com/spell=26654/sweeping-strikes#used-by-npc . 
It seems the ID has been floated around for the same ability, just diff uses, but the wotlk DB and classic DB have the same spell details, and sound for that ID. 

I'd be fine with just fixing the ability to proc from whites, and having WW only consume 1 charge on cast (given it hits at least 2 npcs). It's still not clear if all abilities, or only AoE abilities allow for the SS charge to strike behind the player or outside a 180 frontal cone. Cleave and BS are working fine with it atm, as BS only uses 1 charge, and cleave consumes 2 given 2 mobs are next to eachother.
I'd also like to point out that thunder clap is bugged like whirlwind too. Consumes as many charges as mobs it hits. I think it's suppose to act in conjunction with other aoe abilities, and only consume 1 charge. 
<a href="https://imgur.com/gaKBudC"><img src="https://i.imgur.com/gaKBudC.png" title="source: imgur.com" /></a>

**Comment by bonfax4** [comment]
Disregard the date. Wotlk db clearly states it's a warr related ability. You can also check all warr related abilities and see there isn't any weird one that doesnt fit and comes from npc's. There has to be reason for that.

If they fix WW proccing 4 charges they should also make WW proccing only once per charge. Let's no go half way and make warrior aoe too op.

**Comment by bonfax4** [comment]
@Cooties-cc Was recently looking into this ability again and I was not wrong on sweeping strikes having 2 different id's. Not an npc ability.  https://classic.warcraftlogs.com/reports/7TPgx3Z2vaj1W9QL#fight=9&type=damage-done&source=36
